nova/scheduler/filters/vm_size_threshold_filter.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2021 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from oslo_log import log as logging
+
+from nova.scheduler import filters
+from nova.scheduler.mixins import HypervisorSizeMixin
+
+LOG = logging.getLogger(__name__)
+
+
+class VmSizeThresholdFilter(filters.BaseHostFilter, HypervisorSizeMixin):
+    """Make sure we schedule too big VMs not onto too small hypervisors.
+
+    Any VM bigger or equal to VM_SIZE_THRESHOLD_MB will not be able to spawn on
+    a hypervisor smaller than HV_SIZE_THRESHOLD_MB.
+    """
+
+    VM_SIZE_THRESHOLD_MB = 64 * 1024
+    HV_SIZE_THRESHOLD_MB = 1500 * 1024   # ~ 1.5 TiB
```

Comment:
Those will become config values, I presume?

Reply:
They can, if we want to. They don't have to if we're happy with them ;)